### PR TITLE
Use GitHub CI

### PR DESCRIPTION
**Pull request by Zeitsperre**
### Pull Request Checklist:
- [ ] This PR addresses an already opened issue (for bug fixes / features)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
- [x] `bumpversion (major / minor / patch)` has been called on this branch
- [ ] Tags have been pushed (`git push --tags`)

* **What kind of change does this PR introduce?** 

This implements a Linux-only set of build checks using the GitHub CI actions. This is loosely based off of the recipes implemented in other supported projects (via @cehbrecht, thanks!). The builds here replicate those that we had working previously and leverages `tox` for all stages. Assuming that everything works relatively well out-of-the-box, we can leverage this for a handful of projects with a limited number of builds.

* **Does this PR introduce a breaking change?** 

No.

* **Other information**:

https://github.com/roocs/clisops/pull/124#event-4331983475

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8]
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install packages
+      run: |
+        sudo apt-get -y install pandoc
+      if: matrix.python-version == 3.6
```
Comment (tlvu) [inline .github/workflows/main.yml:17]: how should I read this syntax format?  Install `pandoc` only for python 3.6?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
```
Comment (tlvu) [inline .github/workflows/main.yml:19]: I guess these `uses` are basically means calling a script, in this case the script is `actions/setup-python` at version `v2?

File: .github/workflows/main.yml (hunk 1 above)
Comment (tlvu) [reply .github/workflows/main.yml:17]: Is the indentation wrong?  Compared to other `if` below?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    - name: Lint
+      run: make lint
+      if: matrix.python-version == 3.6
+    - name: Test with pytest
+      run: |
+        pytest --cov xclim -m "not slow"
+    - name: Coveralls
+      uses: coverallsapp/github-action@master
+      with:
+        github-token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (tlvu) [inline .github/workflows/main.yml:37]: Where is this token defined?

**Review by tlvu — APPROVED** [review]
LGTM, just a few questions.

File: .github/workflows/main.yml (hunk 2 above)
Comment (Zeitsperre) [reply .github/workflows/main.yml:19]: That's exactly it.

File: .github/workflows/main.yml (hunk 1 above)
Comment (Zeitsperre) [reply .github/workflows/main.yml:17]: Exactly. This `if` is for GitHub actions, whereas the `if`'s below are for the shell.

File: .github/workflows/main.yml (hunk 3 above)
Comment (Zeitsperre) [reply .github/workflows/main.yml:37]: It's generated on as-needed basis: https://docs.github.com/en/actions/reference/authentication-in-a-workflow#permissions-for-the-github_token. Each job generates a token and once completed, that token is destroyed. Coveralls can therefore not accept more runs for this build once this build has run to completion.

**Review by aulemahal — APPROVED** [review]
Nice! 
For the doctests and slow tests, I think I would suggest removing the normal tests from the `doctests` suite and running the slow tests in  `py38`. Just to reduce the load a bit more and have it a bit more consistent with the naming?

EDIT : And do we need the  `docs` env if we keep the RTD build?

**Comment by Zeitsperre** [comment]
This is good to merge. Thanks for the reviews @aulemahal @tlvu

**Pull request merged**